src/api/entities/AuditTrail.ts

```diff
@@ -0,0 +1,55 @@
+import { Model } from 'objection';
+
+import { BaseModel } from './BaseModel';
+import { ModelObjectOpt } from './ModelObjectOpt';
+
+export enum SharingAction {
+  Add = 'add',
+  Delete = 'delete',
+}
+
+export enum AuditTrailEvents {
+  UpdateSharingPermissions = 'UpdateSharingPermissions',
+}
+
+export type AuditTrailEventData = UpdateSharingPermissionEventData | string;
```

Comment:
I realise that when we retrieve json column from db, it will not convert it into an object, but keep it in string. Maybe provide a get method here?